### PR TITLE
Dsl/dsl-parser module refactor

### LINKED ISSUE #1421: dsl: refactor `dsl_parser.py` and `to_wdl` into multiple modules
Currently, new Chiventure features all must be added to these two files and could potentially make these files very long and convoluted in the future. Therefore, we want to divide these files into modules that are more specific to each feature, and create subdirectories for conversion to intermediate dictionary form, and for conversion from the dictionary to WDL/JSON format. (ie. NPC, rooms, items, etc.) 
team members: Jay, Alex, Andi

### PR DESCRIPTION
**Pull request by awliu2**
We first created a new subdirectory, 'src/dsl_parser'. In this subdirectory, we have created three new files, 'dsl_to_dict.py', 'transformers.py', and 'simple_modifiers.py'. 

'dsl_to_dict.py' specifically focuses on the outward facing export_dict function that is used by the to_wdl module to convert a dictionary/tree into JSON format. 

We moved all of the logical transformers into separate files in order to keep our main outward facing 'dsl_to_dict.py' module clean and concise.

'transformers.py' contains all definitions and functions to transform rooms, actions, items, and the game. For now, we put everything in this file, but as DSL support for Chiventure features grow, it would make sense to split up this file in the future.

'simple_modifiers.py' simply contains all of the one line functions that run simple transformations, i.e labeling items with their id.

Everything compiles and runs as expected.

We have moved each class definition to each file so that in the future, when DSL implements supports features such as NPCs, Battles, etc, they will be placed in separate files. The goal of this is to separate the high-level parsing/conversion function, 'parsed_dict_to_json()', from the rest of the code (which can be hidden at a lower level). Everything has been tested with the example dsl files and works as expected.

### CODE REVIEW

File: src/dsl/src/dsl_parser/dsl_to_dict.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    if debug and "dsl-tree" in debug_modes:
+        print(tree.pretty())
+    intermediate = TreeToDict().transform(tree)
+    if debug and "intermediate" in debug_modes:
+        print(json.dumps(intermediate, indent=2))
+    return intermediate
+
+
+
+class TreeToDict(Transformer):
```
Comment (baigh1) [inline src/dsl/src/dsl_parser/dsl_to_dict.py:38]: These have become super concise and readable! Nice job.

File: src/dsl/src/dsl_parser/dsl_to_dict.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    room = trans.transform_room
+    
+    connections = trans.transform_connections
+
+    item = trans.transform_item
+
+    action = trans.transform_action
+
+    misplaced_property = trans.transform_misplaced
+
```
Comment (baigh1) [inline src/dsl/src/dsl_parser/dsl_to_dict.py:53]: These have become super concise and readable. Very helpful in abstracting information for other users. (Lines 42-52 in dsl_to_dict.py)

File: src/dsl/src/dsl_parser/dsl_to_dict.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        action_names = [name for _, name in s]
+        return ("action_ids", action_names)
+
+    def location(self, s: list[str]) -> tuple[str, str]:
+        """Labels the location as a location"""
+        return ("location", s[0])
+
+    def phrase(self, s: list[Token]) -> str:
+        """Joins a list of strings contained within token objects"""
+        return ' '.join(s)
```
Comment (baigh1) [inline src/dsl/src/dsl_parser/dsl_to_dict.py]: For lines 55-87, I am confused why they are still included indsl_to_dict.py, especially when they are followed by what seems like their counterparts in transfoormations.

File: src/dsl/src/dsl_parser/transformers.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                property_name = property_name.strip()
+                
+                actions_dictionary[action][property_name] = value
+        if matched == False:
+            warn(f"Unexpected object under actions will be ignored: {action_property}")
+    return ("actions", actions_dictionary)
+
+
+def transform_misplaced(self, s: list[Token]) -> str:
+    raise Exception('"property FOR object" syntax is not yet supported')
```
Comment (baigh1) [inline src/dsl/src/dsl_parser/transformers.py:102]: I like overall how you all have split up lower level structures into one file and higher-level structures (player classes, rooms, items, etc) into another. Overall, I think the refactoring was done nicely, just address the redundancy in the dsl_to_dict.py file.

**Review by baigh1 — CHANGES_REQUESTED** [review]

**Comment by alexlee0422** [comment]
Fixed the redundancy error in dsl_to_dict.py

**Review by baigh1 — APPROVED** [review]

**Comment by jacknugent1529** [comment]
Running the standard commands yields an error for me, so something seems to have gone wrong with refactoring
```
python src/parse.py examples/dsl/demo.dsl
 src/parse.py:82	UserWarning: No output file provided. Default to '/Users/jacknugent/Desktop/220CS/chiventure/src/dsl/examples/wdl/demo.wdl' 
Traceback (most recent call last):
  File "/Users/jacknugent/.pyenv/versions/3.9.0/lib/python3.9/site-packages/lark/visitors.py", line 93, in _call_userfunc
    return f(children)
  File "/Users/jacknugent/Desktop/220CS/chiventure/src/dsl/src/dsl_parser/transformers.py", line 22, in transform_game
    game_dictionary = dict((k, v) for k, v in s if k != "ROOM")
  File "/Users/jacknugent/Desktop/220CS/chiventure/src/dsl/src/dsl_parser/transformers.py", line 22, in <genexpr>
    game_dictionary = dict((k, v) for k, v in s if k != "ROOM")
TypeError: cannot unpack non-iterable Tree object
```

**Comment by awliu2** [comment]
Fixed a small typo on line 59 of 'dsl_to_dict.py' (that made it so that the game method was not defined). Code works as expected now, and generates wdl files.

**Review by jacknugent1529 — APPROVED** [review]
Looks good to me! The structure is much clearer than previously

**Comment by jacknugent1529** [comment]
Issue Score: E

Comments: Nice work on this PR! The opening summary was very detailed and well-written. When making refactoring changes, be sure to test the code as rigorously as possible (better tests help with this) to ensure you don't accidentally break it.

**Pull request merged**